packages/cozy-clisk/src/launcher/saveFiles.js

```diff
@@ -125,14 +134,9 @@ const saveFiles = async (client, entries, folderPath, options) => {
       saveOptions.log('warn', '', 'Missing filename property, entry is ignored')
       continue
     }
-    const folderPath = await getOrCreateDestinationPath(
-      client,
-      entry,
-      saveOptions
-    )
     entry = await saveFile(client, entry, {
       ...saveOptions,
-      folderPath
+      folderMap
```

Comment:
I pass the folderMap here to be able to modify it in case we have to create the folder on error in `createFileWithFolderOnError`. If we create a folder, we have to update the folderMap

Reply:
Can you explain the goal of `folderMap` please? It seems to me that this is an anti-pattern (you edit the folderMap from methods so your methods have side-effect and it's not wanted). Why do we need it? 

If this is to be sure that the `path` is created, you should only use https://github.com/cozy/cozy-client/blob/master/packages/cozy-stack-client/src/FileCollection.js#L865-L871. So the need is only to have the `finalPath` (aka `folderPath`) when calling `createFileWithFolderOnError` right?

Reply:
This is exactly the point of this PR. I don't want to call `ensureDirectoryExists` or `createDirectoryByPath` for each file because it can take a lot of time (multiplied by the number of files to save).

Since :
- the final `client.save` of the file needs a `dirId` attribute to specify in which folder the file will be saved, 
- each file can have a different destination folder (because of the `subPath` attribute)
- the user could remove a destination folder during the execution of the konnector

I need to : 

- keep an association of paths to their id : the folderMap, which avoids a lot of calls to CozyClient
- update this folderMap if a destination folder has been removed during the konnector execution to have the new dirId associated to the destination folder.